toxcore/group_connection.c

```diff
@@ -453,6 +453,7 @@ int gcc_handle_packet_fragment(const GC_Session *c, GC_Chat *chat, uint32_t peer
     gconn = get_gc_connection(chat, peer_number);
 
     if (gconn == nullptr) {
+        free(payload);
```

Comment:
How did you find this?

Reply:
Simply checked the CI logs during one of my recent PRs. Apparently `clang --analyze` has been complaining about this for a while now and without anyone noticing. I have then ran clang locally, generating HTML reports to see the full code traces and confirm the memory leaks for myself, as static analyzers sometimes find false positives -- and it did indeed fine some true positives.

Reply:
They do report false positives sometimes, even the same memory leak type of warning that helped find these -- `unix.Malloc` -- currently reports `./toxcore/list.c:247:5: warning: Use of memory after it is freed [unix.Malloc]` which looks like a false positive to me, so I'm not sure about having them error.

It would be nice if we could somehow make it so that newly found warnings, ones that are absent in the branch we are merging the PR into but are present in the PR branch, would require signing off before a PR can be merged, individually every warning one by one, and it would be even nicer if the HTML code traces generated by the static analyses tools could be viewed right from the PR without having to run the tool locally yourself -- but those are very big asks.

I wonder if treating all warnings as errors and having the PR author rewrite their code such that the static analysis tool doesn't trip over it, even if it's a false positive, is a good idea? Not sure.